.github/workflows/rust.yml

```diff
@@ -0,0 +1,48 @@
+
+name: Rust PR
+
+# Run on pushes to master, and on all pull requests.
+on:
+  push:
+    branches:
+    - master
+  pull_request: {}
+
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
```

Comment:
prefer to use shas to tags, especially for actions not owned by github

Reply:
TIL that you can do that! AFAICT, this means that we won't get minor bugfix patches to the action unless we explicitly update the SHA — what's the reason to prefer this?

Reply:
I don't like CI changing underneath us without being able to opt-in to change. It means things can just break and we won't know why.

More generally, tags open up an angle for someone to be able to run arbitrary code in our CI (i.e. if a maintainer loses control of their credentials).

Reply:
That's fair. I'll change this to use SHAs.